tools/RAiDER/models/hrrr.py

```diff
@@ -134,152 +112,82 @@ def load_weather(self, *args, filename=None, **kwargs):
-            levels = np.flip(levels)
-
-        if lats[0, 0] > lats[-1, 0]:
-            flip_axes.append(1)
-            lats = np.flip(lats, 0)
-            yArr = np.flip(yArr)
-
-        if lons[0, 0] > lons[0, -1]:
-            flip_axes.append(2)
-            lons = np.flip(lons, 1)
-            xArr = np.flip(xArr)
-
-        flip_axes = tuple(flip_axes)
-        if flip_axes:
-            t = np.flip(t, flip_axes)
-            z = np.flip(z, flip_axes)
-            q = np.flip(q, flip_axes)
-
-        # Create output dataset
-        ds_new = xarray.Dataset(
-            data_vars=dict(
-                t= (["level", "y", "x"], t,
-                    {"grid_mapping": "proj"}),
-                z= (["level", "y", "x"], z,
-                    {"grid_mapping": "proj"}),
-                q= (["level", "y", "x"], q,
-                    {"grid_mapping": "proj"}),
-                lats=(["y", "x"], lats),
-                lons=(["y", "x"], lons),
-            ),
-            coords=dict(
-                levels=(["level"], levels,
-                       {"units": "millibars",
-                        "long_name":  "pressure_level",
-                        "axis": "Z"}),
-                x=(["x"], xArr,
-                   {"standard_name": "projection_x_coordinate",
-                    "units": "m",
-                    "axis": "X"}),
-                y=(["y"], yArr,
-                   {"standard_name": "projection_y_coordinate",
-                    "units": "m",
-                    "axis": "Y"}),
-            ),
-            attrs={
-                'Conventions': 'CF-1.7',
-                'Weather_model':'HRRR',
-           }
-        )
-
-        # Write projection of output
-        ds_new["proj"] = int()
-        for k, v in self._proj.to_cf().items():
-            ds_new.proj.attrs[k] = v
-
-        ds_new.to_netcdf(out, engine='netcdf4')
-
-
     def _download_hrrr_file(self, DATE, out, model='hrrr', product='prs', fxx=0, verbose=False):
         '''
         Download a HRRR model
         '''
         ## TODO: Check how Herbie does temporal interpolation
-        # corrected_DT = round_date(DATE, datetime.timedelta(hours=self._time_res))
-        # if not corrected_DT == DATE:
-        #     logger.warning('Rounded given datetime from  %s to %s', DATE, corrected_DT)
-        corrected_DT = DATE
-
         H = Herbie(
-            corrected_DT.strftime('%Y-%m-%d %H:%M'),
+            DATE.strftime('%Y-%m-%d %H:%M'),
             model=model,
             product=product,
+            fxx=fxx,
             overwrite=False,
             verbose=True,
             save_dir=Path(os.path.dirname(out)),
         )
-        pf = H.download(":(SPFH|PRES|TMP|HGT):", verbose=verbose)
 
-        self._makeDataCubes(pf, out)
+        # Iterate through the list of datasets
+        #NOTE: this will need to be modified once HERBIE incorporates the
+        # xarray datatree: https://github.com/blaylockbk/Herbie/issues/146
```

Comment:
need to be updated or can be improved in future. The statement makes it sound like its required and will break it here.